app/templates/results.html

```diff
@@ -60,6 +60,7 @@ <h2 class="text-lg">Extracted Images:</h2>
         <canvas id="canvas{{ loop.index }}" width="500" height="300" class="z-10"></canvas>
         <img id="image{{ loop.index }}" src="{{ image }}" alt="Extracted Image" class="absolute top-0 left-0 w-full h-full z-0">
         <input type="text" id="text{{ loop.index }}" class="mt-2 text-input" data-canvas="canvas{{ loop.index }}" style="position: absolute; top: 0; left 0; width: 100%; height: 100%; z-index: 20; background: rgba(255, 255, 255, 0.5); font-size: 16px;">
+        <button onclick="downloadImage('canvas{{ loop.index }}')" class="mt-4 bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded">Download</button>
```

Comment:
Where is this function coming from? How does the canvas work? Does it matter that the image and the input are outside of it?

Reply:
The `downloadImage` function mentioned in the comment is intended to handle the downloading of the canvas content (which includes the image and the overlayed text) as an image file. However, it seems I forgot to include the implementation of this function in the JavaScript section of the HTML template. I will correct this oversight by adding the necessary JavaScript code to handle the canvas and facilitate the download.

Regarding the canvas setup:
- The canvas is used to draw the image and the overlayed text programmatically.
- The image and the input are positioned absolutely to overlay the canvas, but they do not affect the canvas content directly. The canvas content (image and text) needs to be managed through JavaScript to reflect changes made via the input field.

I will add the JavaScript function `downloadImage` to handle the conversion of the canvas content to an image file that can be downloaded, and ensure it integrates correctly with the existing HTML and JavaScript setup. Let's proceed with this amendment.